feat: exposing request from event in interaction handler in order to Ack event with custom payload

Vladmel1234 opened the pull request:
this PR addressing https://github.com/slack-io/slacker/issues/22

Vladmel1234 committed:
feat: exposing request from event in interaction handler

Vladmel1234 committed:
feat: self ack option for interactive events in for more interactivity in view

Vladmel1234 commented:
There is no much sense in first commit without the second one

arusso commented:
I'll fix the workflow error, the line length shouldn't prevent progression.

I'll try and look at this later today. I have questions but need to dig in a bit to make sure they're not self explanatory.

Vladmel1234 commented:
btw if it is needed to add an example i will happily add one

raed-shomali commented:
Examples are always welcome!

raed-shomali commented:
@arusso Let us know when you have had the chance to review the PR.

Vladmel1234 committed:
Interction example (#1)

* fix: New example of adding validation to input and showing error interactivly

In new example there validation of input added, if input is not valid
error showing in the slack view, in order to view the error messege we
need to Ack the messege manually this is the reson for adding client to
the interaction handler,
also fixed errors in all other examples, provided request which is now
mandatory in all interaction handlers

* adding new block suggestion event

---------

Co-authored-by: Vlad Melnik <vlad.m@panorays.com>

Vladmel1234 committed:
fix: adding suggestion interaction type to the interaction event handler
